kv/kvnemesis: TestKVNemesisMultiNode failed

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13631090?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13631090?buildTab=artifacts#/) on master @ [3c837c2a86165188a26e8629b7e21b7ae0fb56ae](https://github.com/cockroachdb/cockroach/commits/3c837c2a86165188a26e8629b7e21b7ae0fb56ae):

Fatal error:

```
panic: r83/3: closed timestamp regression 1705839439.096097964,0 -> 1705839439.073410111,0 (cmd "50112c9caca21e11" index 15 term 6 (LAI=2 CTS=1705839439.073410111,0))
data: 0050112c9caca21e1130066a12420a08ea8cd387f4fd96d61792010360c80372ce020acb020000000000000000020000000139017a6b12016b12ec123339623264633835356536383439363500ffff0100ff01726473630001000361c04e82150e46a393a099eefc539d45126a0a520a1061c04e82150e46a393a099eefc539d451a1e016b12ec123339623264633835356536383439363500ff010001726473632a0a08ea8cd387f4fd96d61730b2bc2938014a0a08ea8cd387f4fd96d6175002120c08ea8cd387f4fd96d61710001800200c286e48010128016b12ec123339623264633835356536383439363500ff010001726473630017ac5bef40f4c66a096e00000002650a00033c120b0308531214ec123339623264633835356536383439363500011a14ec126334306533356439326264656537636500012208080410041801200022080803100318022000220808021002180320002804302f3a1008ffffffffffffffff7f10ffffffff077a008201260a12637264622d7472616365722d7370616e69641210313436396237336661636431383437328201270a13637264622d7472616365722d747261636569641210376130636161346337323237333965378201080a0372656312017620028a010a08bff8c98ee9fd96d617 [recovered]
	panic: r83/3: closed timestamp regression 1705839439.096097964,0 -> 1705839439.073410111,0 (cmd "50112c9caca21e11" index 15 term 6 (LAI=2 CTS=1705839439.073410111,0))
data: 0050112c9caca21e1130066a12420a08ea8cd387f4fd96d61792010360c80372ce020acb020000000000000000020000000139017a6b12016b12ec123339623264633835356536383439363500ffff0100ff01726473630001000361c04e82150e46a393a099eefc539d45126a0a520a1061c04e82150e46a393a099eefc539d451a1e016b12ec123339623264633835356536383439363500ff010001726473632a0a08ea8cd387f4fd96d61730b2bc2938014a0a08ea8cd387f4fd96d6175002120c08ea8cd387f4fd96d61710001800200c286e48010128016b12ec123339623264633835356536383439363500ff010001726473630017ac5bef40f4c66a096e00000002650a00033c120b0308531214ec123339623264633835356536383439363500011a14ec126334306533356439326264656537636500012208080410041801200022080803100318022000220808021002180320002804302f3a1008ffffffffffffffff7f10ffffffff077a008201260a12637264622d7472616365722d7370616e69641210313436396237336661636431383437328201270a13637264622d7472616365722d747261636569641210376130636161346337323237333965378201080a0372656312017620028a010a08bff8c98ee9fd96d617

```
Stack: 

```
goroutine 12914657 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1f1c585?, {0x71f6830, 0xc017ecaae0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:230 +0x65
panic({0x48620c0?, 0xc020a1a010?})
	GOROOT/src/runtime/panic.go:914 +0x21f
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*rangeAsserter).apply.func1({0x52ea70b?, 0xc015799110?}, {0xc0064219c8?, 0x8?, 0xc005dc80a0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/asserter.go:250 +0x165
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*rangeAsserter).apply(0xc009d0d630, 0x3, {0xc0178e0cd8, 0x8}, {0x6, 0xf, 0x0, {0xc011f27e00, 0x1df, 0x1ee}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/asserter.go:301 +0x5d5
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Asserter).Apply(0x3?, 0x0?, 0x103fec8?, {0xc0178e0cd8, 0x8}, {0x6, 0xf, 0x0, {0xc011f27e00, 0x1df, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/asserter.go:231 +0x8d
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.kvnemesisTestCfg.testClusterArgs.func7({{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, {0x17ac5bef40f4c66a, 0x0}, {0x0, ...}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:148 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.replicaApplyTestingFilters({0xc005dc8108?, 0x71f6830?}, 0xc013d54cf0?, 0xc00dd161b8?, {0xc005dc8158?, 0xc00ef312c0?, 0x0?}, 0x30?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_application_state_machine.go:117 +0x1f7
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaAppBatch).Stage(0xc005dc8108, {0x71f6830, 0xc013d54cf0}, {0x7216c80?, 0xc00dd161b8})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_app_batch.go:113 +0xe9
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.mapCmdIter({0x7225ef0?, 0xc005dc84b0}, 0xc006422568)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/cmd.go:192 +0x148
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).applyOneBatch(0xc006422ad8, {0x71f6830, 0xc013d54cf0}, {0x7225ef0, 0xc005dc8480})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:284 +0x150
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).ApplyCommittedEntries(0xc006422ad8, {0x71f6830, 0xc013d54cf0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:251 +0x95
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:1059 +0x1439
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(_, {_, _}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:744 +0x1b0
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc00f7c1500, 0x71f6830?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:689 +0x12b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0xc012313dc0, {0x71f6830, 0xc017ecaae0}, {0x71da820, 0xc00f7c1500}, 0xc0126f6000)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:418 +0x18d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x71f6830?, 0xc017ecaae0?})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x46
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x13a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 12912769
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x415
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisMultiNode616367482
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:327: seed: 2060808370617280199
    kvnemesis_test.go:234: kvnemesis logging to /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/kvnemesis2914343920
```
</p>
</details>

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35460

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/3ae4c75f-03bf-474e-9fe7-d7c024fbe0dc?testReportRun=16&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [45cd83b6838bd515c273955c90ade06ca9fa8794](https://github.com/cockroachdb/cockroach/commits/45cd83b6838bd515c273955c90ade06ca9fa8794):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (4m26s)

```
Stack: 

```
goroutine 116307 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode4133440454
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:327: seed: 1414879219556066563
    kvnemesis_test.go:234: kvnemesis logging to /var/lib/engflow/worker/work/2/exec/_tmp/9f347bcc220fe2eb2bce3feb3eee4e9b/kvnemesis1337112156
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=16</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/d1428c47-fe92-44aa-96ca-1bbdf3367667?testReportRun=13&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e](https://github.com/cockroachdb/cockroach/commits/b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (4m29s)

```
Stack: 

```
goroutine 122051 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode4016196915
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:328: seed: 1232587785184401143
    kvnemesis_test.go:235: kvnemesis logging to /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_13_of_30/test.outputs/kvnemesis3219172245
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=13</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/fcf3cfe4-e213-4b7c-9ace-6a1835bb9981?testReportRun=22&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [1d2534b831775dd84edcf7372d38e35ed4903278](https://github.com/cockroachdb/cockroach/commits/1d2534b831775dd84edcf7372d38e35ed4903278):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (4m26s)

```
Stack: 

```
goroutine 106527 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode1650467610
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:328: seed: 679322385893171862
    kvnemesis_test.go:235: kvnemesis logging to /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_22_of_30/test.outputs/kvnemesis932570081
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/a9af0607-cd19-4a04-933a-fbf38aed930f?testReportRun=6&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [bf013ea0a5311726e65d37e8f047ce39ea2d5f10](https://github.com/cockroachdb/cockroach/commits/bf013ea0a5311726e65d37e8f047ce39ea2d5f10):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (4m27s)

```
Stack: 

```
goroutine 106009 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode2437113376
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:328: seed: 6125298708247058121
    kvnemesis_test.go:235: kvnemesis logging to /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_6_of_30/test.outputs/kvnemesis4249433769
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119681 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-3 T-kv branch-release-23.1]
- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/568f1cdd-55b7-4b7b-8c87-2decc2a46b98?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [d400c11ae174436bfd58137951974168e588f397](https://github.com/cockroachdb/cockroach/commits/d400c11ae174436bfd58137951974168e588f397):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (3m33s)

```
Stack: 

```
goroutine 119513 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode2663985570
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:328: seed: 7324534503789872261
    kvnemesis_test.go:235: kvnemesis logging to /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_9_of_30/test.outputs/kvnemesis3241230119
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119681 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-3 T-kv branch-release-23.1]
- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/186af43c-1026-4b6e-94f5-4c18fb79d6f3?testReportRun=19&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestKVNemesisMultiNode (4m15s)

```
Stack: 

```
goroutine 74014 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestKVNemesisMultiNode3293689279
    test_log_scope.go:81: use -show-logs to present logs inline
    kvnemesis_test.go:328: seed: 927737210348841762
    kvnemesis_test.go:235: kvnemesis logging to /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_19_of_30/test.outputs/kvnemesis4019232538
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=19</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119681 kv/kvnemesis: TestKVNemesisMultiNode failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-23.1]
- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/5b65eb98-ea00-4935-9667-b112196a8e83?testReportRun=19&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [0d3c5b71415de867fedde97de01486d424209848](https://github.com/cockroachdb/cockroach/commits/0d3c5b71415de867fedde97de01486d424209848):


```
          txn.GetForUpdateGuaranteedDurability(ctx, tk(18111364955495334893)) // @1712206364.308315795,0 (<nil>, <nil>)
          txn.Get(ctx, tk(9170750628614985789)) // @1712206364.308315795,0 (<nil>, <nil>)
          txn.GetForUpdate(ctx, tk(5955679492689883723)) // @1712206364.308315795,0 (<nil>, <nil>)
          txn.GetForUpdateSkipLockedGuaranteedDurability(ctx, tk(1095187234670546448)) // @1712206364.308315795,0 (<nil>, <nil>)
          txn.DelRange(ctx, tk(4981749059214929073), tk(14461839080378498375), true /* @s40 */) // QueryIntent request for lock at sequence number 1 but sequence number is ignored [{0 1}]
          txn.Del(ctx, tk(7518199578312411797) /* @s41 */) // omitted
          {
            b := &kv.Batch{}
            b.GetForUpdate(tk(13336625107550524099)) // omitted
            txn.Run(ctx, b) // omitted
          }
          txn.GetForUpdate(ctx, tk(16612592731770747011)) // omitted
          txn.CreateSavepoint(ctx, 15) // omitted
          {
            b := &kv.Batch{}
            b.ScanForUpdate(tk(1859048948803698597), tk(5931568944281215035)) // omitted
            txn.Run(ctx, b) // omitted
          }
          txn.ScanForUpdate(ctx, tk(4255468807830227916), tk(8253637249437608015), 0) // omitted
          txn.ReverseScanForUpdate(ctx, tk(1148492014708157823), tk(13100651418177005298), 0) // omitted
          b := &kv.Batch{}
          txn.CommitInBatch(ctx, b) // omitted
          return nil
        }) // QueryIntent request for lock at sequence number 1 but sequence number is ignored [{0 1}]: QueryIntent request for lock at sequence number 1 but sequence number is ignored [{0 1}]
    kvnemesis.go:178: error applying x.DelRange(ctx, tk(4981749059214929073), tk(14461839080378498375), true /* @s40 */) // QueryIntent request for lock at sequence number 1 but sequence number is ignored [{0 1}]: QueryIntent request for lock at sequence number 1 but sequence number is ignored [{0 1}]
    kvnemesis.go:213: failures(verbose): /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_19_of_25/test.outputs/kvnemesis1867772667/failures
        repro steps: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_19_of_25/test.outputs/kvnemesis1867772667/repro.go
        rangefeed KVs: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_19_of_25/test.outputs/kvnemesis1867772667/kvs-rangefeed.txt
        scan KVs: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_19_of_25/test.outputs/kvnemesis1867772667/kvs-scan.txt
    kvnemesis_test.go:356: [0] proposed: 2442
    kvnemesis_test.go:358: [0] reproposed unchanged: 65
    kvnemesis_test.go:360: [0] reproposed with new LAI: 4
    kvnemesis_test.go:356: [1] proposed: 32
    kvnemesis_test.go:358: [1] reproposed unchanged: 0
    kvnemesis_test.go:360: [1] reproposed with new LAI: 0
    kvnemesis_test.go:356: [2] proposed: 78
    kvnemesis_test.go:358: [2] reproposed unchanged: 0
    kvnemesis_test.go:360: [2] reproposed with new LAI: 0
    kvnemesis_test.go:356: [3] proposed: 69
    kvnemesis_test.go:358: [3] reproposed unchanged: 1
    kvnemesis_test.go:360: [3] reproposed with new LAI: 0
    kvnemesis_test.go:365: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:365
        	            				github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:305
        	Error:      	Should be zero, but was 2
        	Test:       	TestKVNemesisMultiNode
        	Messages:   	kvnemesis detected failures
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestKVNemesisMultiNode2906079865
--- FAIL: TestKVNemesisMultiNode (10.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=19</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119681 kv/kvnemesis: TestKVNemesisMultiNode failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-23.1]
- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/3e3929f1-ea3e-4d9c-9c1d-c422661ea27c?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZuZW1lc2lzOmt2bmVtZXNpc190ZXN0) on master @ [0d3c5b71415de867fedde97de01486d424209848](https://github.com/cockroachdb/cockroach/commits/0d3c5b71415de867fedde97de01486d424209848):


```
          txn.Scan(ctx, tk(5942702328648395716), tk(12231128889100259609), 0) // @1712218528.785843941,0 <nil>
          txn.ReverseScanSkipLocked(ctx, tk(5816083062832197578), tk(10561486304219406719), 0) // @1712218528.785843941,0 <nil>
          txn.ReverseScanForShareSkipLocked(ctx, tk(12524786289612012554), tk(18142357218861825417), 0) // @1712218528.785843941,0 <nil>
          txn.Del(ctx, tk(8194567563964571375) /* @s5 */) // @1712218528.785843941,0 <nil>
          txn.Put(ctx, tk(8194567563964571375), sv(6)) // @1712218528.785843941,0 <nil>
          txn.Scan(ctx, tk(1805268048747277556), tk(10200302720708336125), 0) // @1712218528.785843941,0 (/Table/100/"71b8f3c8620bdeef":v6, <nil>)
          txn.GetForShareSkipLocked(ctx, tk(8194567563964571375)) // @1712218528.785843941,0 (v6, <nil>)
          txn.ReverseScanForUpdateSkipLockedGuaranteedDurability(ctx, tk(10719436284332356064), tk(11355323933532366825), 0) // @1712218528.785843941,0 <nil>
          txn.ReverseScan(ctx, tk(2990563424522545285), tk(7790706350234517601), 0) // @1712218528.785843941,0 <nil>
          txn.CreateSavepoint(ctx, 9) // <nil>
          txn.Del(ctx, tk(3048599556654258201) /* @s7 */) // @1712218528.785843941,0 <nil>
          txn.RollbackSavepoint(ctx, 9) // <nil>
          {
            b := &kv.Batch{}
            b.GetForShare(tk(3048599556654258201)) // (<nil>, <nil>)
            b.GetForShare(tk(12395840483679901846)) // (<nil>, <nil>)
            txn.Run(ctx, b) // @1712218528.785843941,0 <nil>
          }
          txn.Get(ctx, tk(4671415629137746509)) // @1712218528.785843941,0 (<nil>, <nil>)
          txn.ScanForUpdate(ctx, tk(14440349148090554394), tk(17106147330785063347), 0) // @1712218528.785843941,0 <nil>
          txn.ReverseScanForUpdateSkipLockedGuaranteedDurability(ctx, tk(4984364577224726308), tk(8757677364442906033), 0) // @1712218528.785843941,0 (/Table/100/"71b8f3c8620bdeef":v6, <nil>)
          txn.GetForShare(ctx, tk(8194567563964571375)) // QueryIntent request for lock at sequence number 4 but sequence number is ignored [{3 4}]
          return nil
        }) // QueryIntent request for lock at sequence number 4 but sequence number is ignored [{3 4}]: QueryIntent request for lock at sequence number 4 but sequence number is ignored [{3 4}]
    kvnemesis.go:178: error applying x.GetForShare(ctx, tk(8194567563964571375)) // QueryIntent request for lock at sequence number 4 but sequence number is ignored [{3 4}]: QueryIntent request for lock at sequence number 4 but sequence number is ignored [{3 4}]
    kvnemesis.go:213: failures(verbose): /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_2_of_3/test.outputs/kvnemesis1267211005/failures
        repro steps: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_2_of_3/test.outputs/kvnemesis1267211005/repro.go
        rangefeed KVs: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_2_of_3/test.outputs/kvnemesis1267211005/kvs-rangefeed.txt
        scan KVs: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvnemesis/kvnemesis_test/run_2_of_3/test.outputs/kvnemesis1267211005/kvs-scan.txt
    kvnemesis_test.go:356: [0] proposed: 2594
    kvnemesis_test.go:358: [0] reproposed unchanged: 50
    kvnemesis_test.go:360: [0] reproposed with new LAI: 9
    kvnemesis_test.go:356: [1] proposed: 35
    kvnemesis_test.go:358: [1] reproposed unchanged: 5
    kvnemesis_test.go:360: [1] reproposed with new LAI: 0
    kvnemesis_test.go:356: [2] proposed: 67
    kvnemesis_test.go:358: [2] reproposed unchanged: 3
    kvnemesis_test.go:360: [2] reproposed with new LAI: 0
    kvnemesis_test.go:356: [3] proposed: 35
    kvnemesis_test.go:358: [3] reproposed unchanged: 6
    kvnemesis_test.go:360: [3] reproposed with new LAI: 0
    kvnemesis_test.go:365: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:365
        	            				github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:305
        	Error:      	Should be zero, but was 2
        	Test:       	TestKVNemesisMultiNode
        	Messages:   	kvnemesis detected failures
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestKVNemesisMultiNode540265265
--- FAIL: TestKVNemesisMultiNode (30.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119681 kv/kvnemesis: TestKVNemesisMultiNode failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-23.1]
- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvnemesis/kvnemesis_test.go
+++ b/pkg/kv/kvnemesis/kvnemesis_test.go
@@ -131,9 +131,10 @@ func (cfg kvnemesisTestCfg) testClusterArgs(
 			log.Infof(context.Background(), "inserting illegal lease index for %s (seen %d times)", roachpb.Key(key), seen[key])
 			// LAI 1 is always going to fail because the LAI is initialized when the lease
 			// comes into existence. (It's important that we pick one here that reliably
-			// fails because otherwise we may accidentally regress the closed timestamp[^1].
+			// fails because otherwise we may accidentally regress the closed timestamp[^1][^2].
 			//
 			// [^1]: https://github.com/cockroachdb/cockroach/issues/70894#issuecomment-1433244880
+			// [^2]: https://github.com/cockroachdb/cockroach/issues/70894#issuecomment-1881165404
 			return 1
 		}
 	}

--- a/pkg/kv/kvserver/replica_proposal_buf.go
+++ b/pkg/kv/kvserver/replica_proposal_buf.go
@@ -103,6 +103,12 @@ type propBuf struct {
 	testing struct {
 		// leaseIndexFilter can be used by tests to override the max lease index
 		// assigned to a proposal by returning a non-zero lease index.
+		//
+		// If this filter is set, the lease index 1 is reserved for out-of-order
+		// proposals, and propBuf will never submit "real" proposals at index 1.
+		// Returning indices > 1 here is generally unsafe because this may cause
+		// unintended proposal reordering and closed timestamp regressions, as in
+		// https://github.com/cockroachdb/cockroach/issues/118017
 		leaseIndexFilter func(*ProposalData) kvpb.LeaseAppliedIndex
 		// insertFilter allows tests to inject errors at Insert() time.
 		insertFilter func(*ProposalData) error
@@ -218,6 +224,16 @@ func (b *propBuf) Init(
 	b.evalTracker = tracker
 	b.settings = settings
 	b.assignedLAI = p.leaseAppliedIndex()
+
+	// Reserve LAI 1 for out-of-order proposals in tests. A bunch of tests
+	// override proposal's LAI to 1, in order to force reproposals. If these
+	// modified proposals race with a "real" proposal with LAI 1, this can cause a
+	// closed timestamp regression.
+	//
+	// https://github.com/cockroachdb/cockroach/issues/70894#issuecomment-1881165404
+	if b.testing.leaseIndexFilter != nil {
+		b.forwardAssignedLAILocked(1)
+	}
 }
 
 // AllocatedIdx returns the highest index that was allocated. This generally

